src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetCommandHandler.cs

```diff
@@ -36,41 +40,113 @@ internal class GetCommandHandler : IGetCommandHandler
         private readonly PowerShellContextService _powerShellContextService;
+        private readonly ILanguageServer _languageServer;
 
-        public GetCommandHandler(ILoggerFactory factory, PowerShellContextService powerShellContextService)
+        private bool _skipPackageManagementCheck;
+
+        public GetCommandHandler(ILoggerFactory factory, PowerShellContextService powerShellContextService, ILanguageServer languageServer)
         {
             _logger = factory.CreateLogger<GetCommandHandler>();
             _powerShellContextService = powerShellContextService;
+            _languageServer = languageServer;
         }
 
         public async Task<List<PSCommandMessage>> Handle(GetCommandParams request, CancellationToken cancellationToken)
         {
             PSCommand psCommand = new PSCommand();
 
             // Executes the following:
-            // Get-Command -CommandType Function,Cmdlet,ExternalScript | Select-Object -Property Name,ModuleName | Sort-Object -Property Name
+            // Get-Command -CommandType Function,Cmdlet,ExternalScript | Sort-Object -Property Name
             psCommand
                 .AddCommand("Microsoft.PowerShell.Core\\Get-Command")
                     .AddParameter("CommandType", new[] { "Function", "Cmdlet", "ExternalScript" })
-                .AddCommand("Microsoft.PowerShell.Utility\\Select-Object")
-                    .AddParameter("Property", new[] { "Name", "ModuleName" })
                 .AddCommand("Microsoft.PowerShell.Utility\\Sort-Object")
                     .AddParameter("Property", "Name");
 
-            IEnumerable<PSObject> result = await _powerShellContextService.ExecuteCommandAsync<PSObject>(psCommand).ConfigureAwait(false);
+            IEnumerable<CommandInfo> result = await _powerShellContextService.ExecuteCommandAsync<CommandInfo>(psCommand).ConfigureAwait(false);
 
             var commandList = new List<PSCommandMessage>();
             if (result != null)
             {
-                foreach (dynamic command in result)
+                foreach (CommandInfo command in result)
                 {
+                    if (!_skipPackageManagementCheck && command.ModuleName == "PackageManagement" && command.Version < new Version(1, 4, 6))
+                    {
+                        _logger.LogDebug("Old version of PackageManagement detected. Attempting to update.");
+
+                        var takeActionText = "Yes";
+                        MessageActionItem messageAction = await _languageServer.Window.ShowMessage(new ShowMessageRequestParams()
+                        {
+                            Message = "You have a version of PackageManagement that causes issues with the PowerShell extension. Would you like to update PackageManagement (You will need to restart the PowerShell extension after)?",
+                            Type = MessageType.Warning,
+                            Actions = new MessageActionItem[]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5230911)